owncloudApp/src/main/java/com/owncloud/android/ui/activity/PassCodeActivity.kt

```diff
@@ -0,0 +1,425 @@
+/*
+ * ownCloud Android client application
+ *
+ * @author Bartek Przybylski
+ * @author masensio
+ * @author David A. Velasco
+ * @author Christian Schabesberger
+ * @author David González Verdugo
+ * @author Abel García de Prada
+ * Copyright (C) 2011 Bartek Przybylski
+ * Copyright (C) 2021 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.ui.activity
+
+import com.owncloud.android.utils.DocumentProviderUtils.Companion.notifyDocumentProviderRoots
+import android.widget.TextView
+import android.widget.EditText
+import android.os.Bundle
+import android.view.WindowManager
+import com.owncloud.android.R
+import android.widget.LinearLayout
+import android.view.View.OnFocusChangeListener
+import android.content.Intent
+import android.preference.PreferenceManager
+import android.text.TextWatcher
+import android.text.Editable
+import android.view.KeyEvent
+import android.view.View
+import android.widget.Button
+import com.owncloud.android.BuildConfig
+import com.owncloud.android.utils.PreferenceUtils
+import timber.log.Timber
+import java.lang.IllegalArgumentException
+import java.lang.StringBuilder
+import java.util.Arrays
+
+class PassCodeActivity : BaseActivity() {
+    private lateinit var mBCancel: Button
+    private lateinit var mPassCodeHdr: TextView
+    private lateinit var mPassCodeHdrExplanation: TextView
+    private lateinit var mPassCodeError: TextView
```

Comment:
When we migrate files from java to Kotlin, we should try to follow the [code conventions of Kotlin](https://kotlinlang.org/docs/coding-conventions.html#function-names). We can remove the initial `m` at the moment